src/coreclr/vm/runtimehandles.cpp

```diff
@@ -2839,7 +2839,7 @@ FCIMPL1(INT32, ModuleHandle::GetMDStreamVersion, ReflectModuleBaseObject * pModu
     REFLECTMODULEBASEREF refModule = (REFLECTMODULEBASEREF)ObjectToOBJECTREF(pModuleUNSAFE);
 
     if (refModule == NULL)
-        FCThrowRes(kArgumentNullException, W("Arg_InvalidHandle"));
```

Comment:
I do not understand why it is a good idea to return `-1` instead of throwing. All other similar methods throw exception on empty handles. What makes this method special that it is better to return invalid version number instead?

@marek-safar @steveharter

Context: https://github.com/dotnet/runtime/issues/72943

Reply:
This rule is in the "Avoid" category that is just a suggestion, it is not a hard don't ever do this. This rule is violated by many property getters throughout the framework, especially the older ones.

Reply:
It depends if we want EmptyHandle to be more like InvalidHandle or not. To me it feels more logical to return a value from EmptyHandle.MDStreamVersion than to throw.

Reply:
We have number of handle structs with throwing properties. For example, `GCHandle.Target` throws when the handle is not initialized.

Reply:
What about special casing the `EmptyHanlde` only? Probably we should keep throwing for other handles.

Reply:
> What about special casing the EmptyHanlde only (within the MDStreamVersion getter, not here)? 

It is what this change does.

Reply:
I meant any other case when `default` is used, now I see `EmptyHanlde` is returning `default` 😆

Reply:
Throwing for `default` handle makes sense to me, I guess we don't want to make `EmptyHanlde` return something other than `default`.  So, should we close the PR and the issue as by design?